web_client/dialogs/editStyleGroups.js

```diff
@@ -0,0 +1,167 @@
+import tinycolor from 'tinycolor2';
+
+import View from 'girder/views/View';
+
+import StyleModel from '../models/StyleModel';
+import editStyleGroups from '../templates/dialogs/editStyleGroups.pug';
+import 'girder/utilities/jquery/girderModal';
+import '../stylesheets/dialogs/editStyleGroups.styl';
+
+/**
+ * Create a modal dialog with fields to edit and create annotation
+ * style groups.
+ */
+const EditStyleGroups = View.extend({
+    events: {
+        'click .h-create-new-style': '_createNewStyle',
+        'click .h-save-new-style': '_saveNewStyle',
+        'change .h-style-def': '_updateStyle',
+        'changeColor .h-colorpicker': '_updateStyle',
```

Comment:
If you try to type a color in the color field, the focus keeps getting snatched away from the input box.  This makes it hard (but not quite impossible) to enter colors via the keyboard (at least on my version of Chrome).

Reply:
It should be fixed now.  I was being overly ambitious about rerendering the view.